### PR TITLE
Migrate to file based configuration- batch score component.

### PR DESCRIPTION
**Pull request by arun-rajora**
This PR migrates the endpoint inferencing component to the file-based configuration- batch score component.

### CODE REVIEW

**Review by iamrk04 — CHANGES_REQUESTED** [review]

File: assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml [hunk 1]
```diff
@@ -38,9 +42,7 @@ inputs:
 
       For AOAI model, the following pattern can be used,
       {
-          "messages":
-          [
-            {"role": "user", "content": "###<prompt>" } ],
+          "messages": ###<prompt>,
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml:45]: Updated it. Thanks!

File: assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml [hunk 2]
```diff
@@ -164,52 +193,67 @@ jobs:
...
-      mini_batch_results_out_directory:
-        type: uri_folder
-      metrics_out_directory:
+      mini_batch_results_output_directory:
         type: uri_folder
     resources:
       instance_count: ${{parent.inputs.instance_count}}
     max_concurrency_per_instance:  ${{parent.inputs.max_concurrency_per_instance}}
-    mini_batch_size: "100KB"
+    mini_batch_size: 100KB #${{parent.inputs.mini_batch_size}}
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml:234]: This is expected. Right now, this binding is not supported. "100KB #" will be removed as soon as binding support is released.

File: assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+      configuration_file: ${{parent.jobs.config_generation.outputs.batch_score_config}}
+    outputs:
+      job_output_path:
+        type: uri_file
+      mini_batch_results_output_directory:
+        type: uri_folder
+    resources:
+      instance_count: ${{parent.inputs.instance_count}}
+    max_concurrency_per_instance:  ${{parent.inputs.max_concurrency_per_instance}}
+    mini_batch_size: 100KB #${{parent.inputs.mini_batch_size}}
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml:238]: Same as previous comment.

File: assets/aml-benchmark/tests/batch_benchmark_inference/test_batch_benchmark_config_generator.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import pytest
+import sys
+import tempfile
+
+from ..test_utils import get_src_dir, get_current_path
+from aml_benchmark.utils.constants import AuthenticationType
+
+
+sys.path.append(get_src_dir())
+print(get_src_dir())
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/tests/batch_benchmark_inference/test_batch_benchmark_config_generator.py:18]: Updated. Thanks!

File: assets/aml-benchmark/components/batch_benchmark_config_generator/asset.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+type: component
+spec: spec.yaml
+categories: ["Benchmarking"]
+test:
+  pytest:
+    enabled: true
+    conda_environment: ../../dev_conda_env.yaml
+    tests_dir: ../../tests
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_config_generator/asset.yaml]: I have updated it. Thanks!

File: assets/aml-benchmark/components/batch_benchmark_config_generator/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    description: Type of model. Can be one of ('aoai', 'oss', 'vision_oss')
+    optional: false
+    enum:
+        - oai
+        - oss
+        - vision_oss
+  authentication_type:
+    type: string
+    optional: False
+    description: Authentication type for endpoint.
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_config_generator/spec.yaml]: Updated. Thanks!

File: assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml [hunk 7]
```diff
@@ -164,52 +193,67 @@ jobs:
...
       ground_truth_metadata:
         type: uri_folder
-  # Inference
-  endpoint_batch_score:
-    type: parallel
-    component: azureml:batch_benchmark_score:0.0.6
+  # Config generation
+  config_generation:
+    type: command
+    component: azureml:batch_benchmark_config_generator:0.0.1
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml]: I have updated it to latest. Thanks!

File: assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+      model_type: ${{parent.inputs.model_type}}
+      app_insights_connection_string: ${{parent.inputs.app_insights_connection_string}}
+    outputs:
+      batch_score_config:
+        type: uri_file
+        path: azureml://datastores/${{default_datastore}}/paths/azureml/${{name}}/${{output_name}}.json
+  # Inference
+  endpoint_batch_score:
+    type: parallel
+    component: azureml://registries/azureml/components/batch_score_llm/labels/latest
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml]: Updated. Thanks!

File: assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+      additional_columns: ${{parent.inputs.additional_columns}}
+    outputs:
+      formatted_data:
+        type: mltable
+      ground_truth_metadata:
+        type: uri_folder
+  # Config generation
+  config_generation:
+    type: command
+    component: azureml:batch_benchmark_config_generator:0.0.1
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml]: Updated. Thanks!

File: assets/aml-benchmark/components/src/aml_benchmark/utils/online_endpoint/endpoint_utils.py [hunk 10]
```diff
@@ -87,16 +87,26 @@ def hash_payload_prompt(
...
-                if isinstance(prompt[0], dict):
-                    prompt = " ".join([p['content'] for p in prompt])
-            elif model.is_aoai_model():
-                if isinstance(payload["messages"], list):
-                    prompt = " ".join([msg.get("content", "") for msg in payload["messages"]])
+            if model.is_oss_model() or model.is_aoai_model():
+                if "input_data" in payload and "input_string" in payload["input_data"]:
+                    # For OSS endpoint
+                    prompt = payload["input_data"]["input_string"]
+                    if isinstance(prompt[0], dict):
```
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/components/src/aml_benchmark/utils/online_endpoint/endpoint_utils.py:94]: Updated. Thanks for catching this!

File: assets/aml-benchmark/tests/batch_benchmark_inference/test_batch_benchmark_config_generator.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arun-rajora) [reply assets/aml-benchmark/tests/batch_benchmark_inference/test_batch_benchmark_config_generator.py:18]: Removing noqa is still showing error.

File: assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (iamrk04) [reply assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml:45]: This isn't valid for `chat_completion`, but for `text_completion`. For `chat_completion`, it should be what it was earlier.

File: assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml [hunk 11]
```diff
@@ -164,52 +193,68 @@ jobs:
...
+    environment_variables:
+      BATCH_SCORE_INITIAL_REQUEST_TIMEOUT: '4'
+      BATCH_SCORE_DELAY_AFTER_SUCCESSFUL_REQUEST: 'False'
+      BATCH_SCORE_MAX_REQUEST_TIMEOUT: '300'
+      
   # Reformat
   batch_output_formatter: 
     type: command
-    component: azureml:batch_output_formatter:0.0.7
+    component: azureml:batch_score_llm@latest
```
Comment (iamrk04) [inline assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml:246]: Looks like component name was replaced by mistake.

File: assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+    compute: ${{parent.inputs.inference_compute}}
+    environment_variables:
+      BATCH_SCORE_INITIAL_REQUEST_TIMEOUT: '4'
+      BATCH_SCORE_DELAY_AFTER_SUCCESSFUL_REQUEST: 'False'
+      BATCH_SCORE_MAX_REQUEST_TIMEOUT: '300'
+      
+  # Reformat
+  batch_output_formatter: 
+    type: command
+    component: azureml:batch_score_llm@latest
```
Comment (iamrk04) [inline assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml:251]: same with component name.

File: assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml (hunk 11 above)
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference/spec.yaml]: I have updated it. Thanks for catching this!

File: assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml (hunk 12 above)
Comment (arun-rajora) [reply assets/aml-benchmark/components/batch_benchmark_inference_with_inference_compute/spec.yaml]: I have updated it. Thanks for catching this!

**Review by iamrk04 — DISMISSED** [review]

**Review by iamrk04 — APPROVED** [review]

**Pull request merged**